react-components/src/components/NodeCacheProvider/NodeCacheProvider.tsx

```diff
@@ -97,8 +98,15 @@ export const useFdmAssetMappings = (
 export function NodeCacheProvider({ children }: { children?: ReactNode }): ReactElement {
   const fdmClient = useFdmSdk();
   const cdfClient = useSDK();
+  const revealKeepAliveData = useRevealKeepAlive();
 
-  const fdmCache = useMemo(() => new FdmNodeCache(cdfClient, fdmClient), []);
+  const fdmCache = useMemo(() => {
+    const cache = revealKeepAliveData?.fdmNodeCache.current ?? new FdmNodeCache(cdfClient, fdmClient);
+    if (revealKeepAliveData !== undefined) {
+      revealKeepAliveData.fdmNodeCache.current = cache;
+    }
+    return cache;
```

Comment:
I think this part is pretty hard to read. Can it be simplified to something like this instead?

```typescript
if (revealKeepAliveData.fdmNodeCache.current === undefined) {
  revealKeepAliveData.fdmNodeCache.current = new FdmNodeCache(..)
}
```

Reply:
Actually, I would prefer to do defensive programming here instead

```typescript
if (revealKeepAliveData.fdmNodeCache.current !== undefined) {
  return;
}

revealKeepAliveData.fdmNodeCache.current = new FdmNodeCache(..)
```

Reply:
None of these will do what we want :sweat: I only want to set `revealKeepAliveData.fdmNodeCache.current` to the new cache if `revealKeepAliveData` itself is present. That check is not done in either of the snippets